common/internal-types/src/channels.rs

```diff
@@ -251,6 +251,19 @@ impl ChannelChange {
     }
 }
 
+// create a newtype, CorruptedChannelEntry, to represent a channel that is corrupted. It's a wrapper around channelentry
+// basically
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct CorruptedChannelEntry {
+    pub channel: ChannelEntry,
```

Comment:
Consider making this private member and add a getter that allows access via an immutable (non-`mut`) reference only.

You can also make this a unit struct (i.e. `pub struct CorruptedChannelEntry(ChannelEntry)`)